python/ray/train/tests/test_new_persistence.py

```diff
@@ -0,0 +1,119 @@
+    for i in range(start, config.get("num_iterations", 5)):
+        time.sleep(0.5)
+
+        with open(f"artifact-{i}.txt", "w") as f:
+            f.write(f"{i}")
+
+        temp_dir = tempfile.mkdtemp()
+        with open(os.path.join(temp_dir, "dummy.pkl"), "wb") as f:
+            pickle.dump({"iter": i}, f)
+
+        train.report({"iter": i}, checkpoint=air.Checkpoint.from_directory(temp_dir))
+
+
+@pytest.mark.parametrize("storage_path_type", [None, "nfs", "cloud", "custom_fs"])
+def test_tuner(monkeypatch, storage_path_type, tmp_path):
+    # Set the cache dir to some temp directory
+    LOCAL_CACHE_DIR = tmp_path / "ray_results"
+    monkeypatch.setenv("RAY_AIR_LOCAL_CACHE_DIR", str(LOCAL_CACHE_DIR))
+
+    context_manager = (
+        mock_s3_bucket_uri if storage_path_type == "cloud" else dummy_context_manager
+    )
+
+    exp_name = "simple_persistence_test"
+
+    with context_manager() as cloud_storage_path:
+        storage_filesystem = None
+        if storage_path_type is None:
+            storage_path = None
+        elif storage_path_type == "nfs":
+            storage_path = str(tmp_path / "fake_nfs")
+        elif storage_path_type == "cloud":
+            storage_path = str(cloud_storage_path)
+        elif storage_path_type == "custom_fs":
+            from fsspec.implementations.dirfs import DirFileSystem
+            from fsspec.implementations.local import LocalFileSystem
+
+            import pyarrow.fs
+
+            storage_path = "mock_bucket"
+            storage_filesystem = pyarrow.fs.PyFileSystem(
+                pyarrow.fs.FSSpecHandler(
+                    DirFileSystem(
+                        path=str(tmp_path / "custom_fs"), fs=LocalFileSystem()
+                    )
+                )
+            )
+
+        NUM_ITERATIONS = 6  # == num_checkpoints == num_artifacts
+        NUM_TRIALS = 2
+        tuner = tune.Tuner(
+            train_fn,
+            param_space={"num_iterations": NUM_ITERATIONS},
+            run_config=train.RunConfig(
+                storage_path=storage_path,
+                storage_filesystem=storage_filesystem,
+                name="simple_persistence_test",
+                verbose=0,
+                failure_config=train.FailureConfig(max_failures=1),
+            ),
+            # 2 samples, running 1 at at time to test with actor reuse
+            tune_config=tune.TuneConfig(
+                num_samples=NUM_TRIALS, max_concurrent_trials=1
+            ),
+        )
+        tuner.fit()
+
+        local_inspect_dir = tmp_path / "inspect"
+        if storage_path:
+            download_from_uri(
+                storage_path,
+                str(local_inspect_dir),
+                source_filesystem=storage_filesystem,
+            )
+        else:
+            local_inspect_dir = LOCAL_CACHE_DIR
+
+    assert len(list(local_inspect_dir.glob("*"))) == 1  # Only expect 1 experiment dir
+    exp_dir = local_inspect_dir / exp_name
+
+    # Files synced by the driver
```

Comment:
Could you also put in the comments a full listing of the files that were generated? This would help with understanding the test cases more easily.